weave/trace/weave_client.py

```diff
@@ -1377,21 +1376,20 @@ def send_obj_create() -> ObjCreateRes:
             return self.server.obj_create(req)
 
         res_future: Future[ObjCreateRes] = self.future_executor.defer(send_obj_create)
-
         digest_future: Future[str] = self.future_executor.then(
             [res_future], lambda res: res[0].digest
         )
 
         ref: Ref
-        if is_opdef:
+        if is_op(orig_val):
             ref = OpRef(self.entity, self.project, name, digest_future)
         else:
             ref = ObjectRef(self.entity, self.project, name, digest_future)
 
         # Attach the ref to the object
         try:
             set_ref(orig_val, ref)
-        except:
+        except Exception:
```

Comment:
blank except is actually `BaseException` which is not safe
https://stackoverflow.com/questions/27995057/why-is-it-recommended-to-derive-from-exception-instead-of-baseexception-class-in